src/heretic/mn_menu.c

```diff
@@ -933,6 +934,25 @@ static void DrawSaveLoadBottomLine(const Menu_t *menu)
     M_snprintf(pagestr, sizeof(pagestr), "PAGE %d/%d", savepage + 1, SAVEPAGE_MAX + 1);
     MN_DrTextA(pagestr, ORIGWIDTH / 2 - MN_TextAWidth(pagestr) / 2, y);
 
+    // [crispy] print "modified" (or created initially) time of savegame file
+    if (SlotStatus[CurrentItPos] && !FileMenuKeySteal)
+    {
+        struct stat st;
+        char filedate[32];
+
+        stat(SV_Filename(CurrentItPos), &st);
+// [FG] suppress the most useless compiler warning ever
+#if defined(__GNUC__)
+#  pragma GCC diagnostic push
+#  pragma GCC diagnostic ignored "-Wformat-y2k"
+#endif
+        strftime(filedate, sizeof(filedate), "%x %X", localtime(&st.st_mtime));
```

Comment:
Woke up with a bad feeling, that it can be unsafe since Raven's font set doesn't have a backslash symbol, and attempt to draw non existing symbol is a direct path to program crash. But looks likes it is safe all around, none of existing [date formats](https://en.wikipedia.org/wiki/List_of_date_formats_by_country) are using `\`.

Reply:
We could just fix font drawing to skip non-existent character patches.

Reply:
Neat idea, even outside of this PR, this can potentially improve compatibility with custom texts. For small font drawing, it should be something like this:

<details>

```diff
+++ crispy-doom/src/heretic/mn_menu.c	Tue Apr 23 10:24:01 2024
@@ -631,6 +631,10 @@ void MN_DrTextA(const char *text, int x, int y)
         {
             x += 5;
         }
+        else if (c > 91)
+        {
+            continue;
+        }
         else
         {
             p = W_CacheLumpNum(FontABaseLump + c - 33, PU_CACHE);

```

</details>

Why exactly `> 91`: no font characters available starting from `92`, acording to [ASCII table](https://upload.wikimedia.org/wikipedia/commons/1/1b/ASCII-Table-wide.svg). Raven fonts starting from `FONTA01`, while in ASCII characters it's a `33`. Last available lump for small font is `FONTA59`, so `59 + 33` equals `92`.

Hopefully. 🙂 At least this way I no longer have crashes by using `\` symbol in text drawing. But probably worth to leave it for separated PR?

Reply:
Yes, Heretic's A-font does only have 59 characters, so everything after `33+59` should be skipped. I would extend the earlier check, though, and just print a blank space.

Reply:
💡! True, patch drawing is guarded by following `else` condition, so it can be simplified to just 
```
        if (c < 33 || c > 91)
        {
            x += 5;
        }
```

Worth to do same trick with big font `FONTB**`, except it have 58 characters, so check should be `if (c < 33 || c > 90)`

Reply:
Yes, please. And I think it belongs into this bug report, because the time and date string is not under our control.